pkg/shell/superuser.jsx

```diff
@@ -176,19 +176,19 @@ export class SuperuserDialogs extends React.Component {
 
     unlock(error) {
         this.superuser.Stop().always(() => {
-            if (this.superuser.Bridges.length == 1) {
-                this.start(this.superuser.Bridges[0], error);
-            } else {
-                this.set_unlock_state({
-                    method: this.superuser.Bridges[0],
-                    methods: this.superuser.Bridges,
-
-                    error: error,
-                    change: val => this.update_unlock_state({ method: val }),
-                    cancel: () => this.set_unlock_state({ closed: true }),
-                    apply: () => this.start(this.state.unlock_dialog_state.method)
-                });
-            }
+            cockpit.spawn(["sudo", "-v", "-n"], { err: "message" })
+                    .fail(e => {
```

Comment:
This is an ES6 code file, please use standard Promise API, i. e. `.catch()` instead of `.fail()`. Also, I think `.then()` needs to go first, otherwise this may behave weird due to promise chaining. I recently had a case like this where the .then handler received the caught error.

Reply:
Yes, I had fun debugging this when switching to `.catch(...)`...  Done in the the new `can_do_sudo` function.